etl-job/fhir_import_export.py

```diff
@@ -221,157 +207,449 @@ def _load_all(hostname,
-                   config_path: str | None = None):
-    """Clear out graph and flat metadata for project """
-    # check permissions
+def _run_subprocess(cmd: List[str], cwd: str, output: Dict[str, Any], error_msg: str) -> bool:
+    """Run subprocess command and handle errors."""
     try:
-        grip_delete(hostname, graph_name=_get_graphName(),
-                    project_id=f"{program}-{project}",
-                    output=output, access_token=_get_token())
-        output['logs'].append(f"EMPTIED graph for {program}-{project}")
+        result = subprocess.run(cmd, cwd=cwd, check=True, capture_output=True, text=True)
+        logging.info(f"Successfully ran command: {' '.join(cmd)}")
+        if result.stdout:
+            output['logs'].append(f"STDOUT: {result.stdout.strip()}")
+        return True
+    except subprocess.CalledProcessError as e:
+        detailed_error = f"{error_msg}. Command failed with return code {e.returncode}."
+        if e.stderr:
+            detailed_error += f"\nGit Error Details:\n{e.stderr.strip()}"
+        _handle_error(output, detailed_error, subprocess.CalledProcessError)
+    except FileNotFoundError:
+        _handle_error(output, f"Command not found: {cmd[0]}", FileNotFoundError)
+    except Exception as e:
+        _handle_error(output, f"An unexpected error occurred: {e}", Exception)
+    return False
+
+
+def _write_output_to_client(output):
+    '''
+    formats output as json to stdout so it is passed back to the client,
+    most importantly to display relevant logs from the job erroring out
+    '''
+    logging.info(f"[out] {json.dumps(output, separators=(',', ':'))}")
 
-        for index in ["researchsubject", "specimen", "file"]:
-            meta_flat_delete(project_id=f"{program}-{project}", index=index)
-        output['logs'].append(f"EMPTIED flat for {program}-{project}")
+
+def _handle_error(output: Dict[str, Any], message: str, exception_type: type = Exception):
+    """A helper function to log errors, update output, and raise an exception."""
+    logging.error(message)
+    output['logs'].append(message)
+    _write_output_to_client(output)
+    raise exception_type(message)
+
+def _validate_and_extract_input(input_data: Dict[str, Any]) -> Dict[str, Any]:
+    """Validate and extract required fields from the input_data dictionary."""
+    required_fields = {
+        'ghUserName': "input data must contain a `ghUserName`",
+        'ghToken': "input data must contain a `ghToken`",
+        'ghCommitHash': "input data must contain a `ghCommitHash`",
+        'ghRepoUrl': "input data must contain a `ghRepoUrl`",
+        'bucketName': "input data must contain a `bucketName`",
+        'profile': "input data must contain a `profile`",
+        'APIEndpoint': "input data must contain a `APIEndpoint`",
+    }
+
+    for field, error_msg in required_fields.items():
+        if field not in input_data or not input_data[field]:
+            raise ValueError(error_msg)
```

Comment:
if  you wanna keep the same error message, you can shorten it

```suggestion
    required_fields = ['ghUserName', 'ghToken', 'ghCommitHash', 'ghRepoUrl', 'bucketName', 'profile', 'APIEndpoint', ]

    for field, error_msg in required_fields.items():
        if field not in input_data or not input_data[field]:
            raise ValueError(f"input data must contain a `{field}`")
```